src/v/storage/compaction_reducers.cc

```diff
@@ -234,36 +234,34 @@ ss::future<ss::stop_iteration> copy_data_segment_reducer::do_compaction(
     using stop_t = ss::stop_iteration;
     auto to_copy = filter(std::move(b));
     if (to_copy == std::nullopt) {
-        return ss::make_ready_future<stop_t>(stop_t::no);
+        co_return stop_t::no;
     }
-    return compress_batch(original, std::move(to_copy.value()))
```

Comment:
🤦 These two diffs are much easier to view on a whitespace-aware viewer, and it makes the reason I split the change into two (for review-ability) clearer.